include/coro/sync_wait.hpp

```diff
@@ -34,7 +34,7 @@ class sync_wait_task_promise_base
 {
 public:
     sync_wait_task_promise_base() noexcept = default;
-    virtual ~sync_wait_task_promise_base() = default;
+    ~sync_wait_task_promise_base()         = default;
```

Comment:
Can you make the base destructor protected so in the future if someone was to try and delete via a base pointer the compiler would error?

Reply:
Already make the destructor protected